Automattic/wp-calypso
# Remove lodash from login selectors

Removing `lodash` from state, particularly `_.get`, will require some level of understanding of each portion of state, so that we avoid falling into the trap of using `?.` everywhere instead of just where it's needed.

I prepared this small PR to get some feedback on my approach, and whether you think it's the right one.

For example, for `login` state, I decided that selectors shouldn't need to able to work with `undefined` state, but rather can expect for state to have been initialised by the base reducer, such that `state.login` is always present and is an object. Is this the right approach?

#### Changes proposed in this Pull Request

* Remove `lodash`, particularly `_.get`, from the login selectors

#### Testing instructions

* Ensure that the login selector tests pass
* Smoke-test login and ensure it continues to work correctly
